src/commands/flex/plugins/deploy.js

```diff
@@ -1,39 +1,141 @@
-const FlexPluginScripts = require('../../../sub-commands/flex-plugin-scripts');
+const { flags } = require('@oclif/command');
+const semver = require('semver');
+const { progress, logger } = require('flex-plugins-utils-logger');
+
+const FlexPlugin = require('../../../sub-commands/flex-plugin');
 const { createDescription } = require('../../../utils/general');
+const { TwilioCliError } = require('../../../exceptions');
 
 /**
  * Builds and then deploys the Flex plugin
  */
-class FlexPluginsDeploy extends FlexPluginScripts {
+class FlexPluginsDeploy extends FlexPlugin {
   constructor(argv, config, secureStorage) {
     super(argv, config, secureStorage, { strict: false });
 
-    this.exit = process.exit;
-    process.exit = (exitCode) => {
-      if (exitCode === 0) {
-        return;
-      }
+    this.scriptArgs = [];
+  }
+
+  /**
+   * Main method
+   */
+  async doRun() {
+    const args = ['--quiet', '--persist-terminal'];
+
+    await progress('Validating plugin deployment', async () => this.validateVersion(), false);
```

Comment:
We need to include the log for mentioning the profile "Using profile dev" or "Using account details from profile: dev"